app/src/main/webapp/js/services.js

```diff
@@ -42,51 +42,84 @@ esbMessageAdminServices.service('EsbMessageService', ['$http', '$q', 'Globals',
     };
 
     self.getSyncKeysTree = function() {
-        return $http({method: 'GET', url: "api/key/tree/Entities"});
+        return self.getTree("Entities");
     };
 
     self.getSearchKeysTree = function() {
-        return $http({method: 'GET', url: "api/key/tree/SearchKeys"});
+        return self.getTree("SearchKeys");
     };
 
-    self.addKey = function(argParentId, argName, argType, argValue) {
+    self.respondSuccess = function(argResponse) {
+        if(argResponse.status==="Success") {
+            return argResponse.tree;
+        } else {
+            messageCenterService.add('danger', argResponse.errorMessage, {status: messageCenterService.status.permanent});
+            return null;
+        }
+    }
+
+    self.respondError= function()  {
```

Comment:
Oh I see, it is more like, `handleSuccess` / `handleError` I guess -- I first thought they all they did was just display alerts.